presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -435,6 +436,31 @@ public PlanWithProperties visitWindow(WindowNode node, StreamPreferredProperties
             return deriveProperties(result, child.getProperties());
         }
 
+        @Override
+        public PlanWithProperties visitDelete(DeleteNode node, StreamPreferredProperties parentPreferences)
+        {
+            if (!node.getInputDistribution().isPresent()) {
+                return visitPlan(node, parentPreferences);
+            }
+            DeleteNode.InputDistribution inputDistribution = node.getInputDistribution().get();
+            StreamPreferredProperties childRequirements = parentPreferences
+                    .constrainTo(node.getSource().getOutputVariables())
+                    .withDefaultParallelism(session)
+                    .withPartitioning(inputDistribution.getPartitionBy());
```

Comment:
do we need to worry about order sensitivity here?

Reply:
I think not. The AddLocalExchanges is to add local exchanges and we do not expect it add sort node to enforce the input order (this will be in a separate optimizer rule, let's say it's DeleteSortRule).
Order property can be used in two ways here:
* if the input is already partially sorted or even totally sorted, we can record it in the DeleteNode so as to instruct the DeleteSortRule to simplify the sort process, if this DeleteSortRule runs after AddLocalExchanges. For this part we need to add more attribute to DeleteNode and I consider it as a future optimization to avoid introducing more complex changes before we see more usage of this new input distribution field
* assert that the input is sorted here if the DeleteSortRule is before AddLocalExchanges. Again I plan to do this assertion in DeleteNode during runtime (and we need it anyway for robustness even if we assert the data is sorted in the plan here, since we may introduce other changes later which break this property) and avoid introducing more complex changes.